client/state/ui/reducer.js

```diff
@@ -74,6 +75,13 @@ export const isPreviewShowing = createReducer( false, {
 		isShowing !== undefined ? isShowing : state,
 } );
 
+// FIXME: We can't detect window size on the server, so until we have more intelligent detection,
+// use 769, which is just above the general maximum mobile screen width.
+export const viewportWidth = createReducer( 769, {
+	[ VIEWPORT_WIDTH_SET ]: ( state, { width } ) =>
+		! isNaN( width ) ? width : state,
+} );
```

Comment:
> I tend to use createReducer even for small reducers
😢 

> as I like its expressiveness

??

```js
const reducer = ( state = 0, { type, value } ) => state + get( {
	[ INCREMENT ]: 1,
	[ DECREMENT ]: -1,
	[ ADD ]: value,
	[ SUB ]: -value,
}, type, 0 );
```

```js
const reducer = ( state = 0, { type, value } ) => {
	switch ( type ) {
		case INCREMENT: return state + 1;
		case DECREMENT: return state - 1;
		case ADD: return state + value;
		case SUB: return state - value;
		default: return state;
	}
}
```

Just curious, what do you find more expressive about it?

> and I like the idea of us converging towards common patterns across Calypso.

Yeah I get this. To me `createReducer()` is confusing and for 95% of the reducers doesn't add anything, plus it's a lot of syntax noise to create the thunks for each type of action, plus it doesn't work too well when we want to listen to more than one action type for a given update.

It's confusing to me because it does stuff behind the scenes that isn't all that clear whereas when we have really simple reducers then what-you-see-is-what you get, plus it's that many fewer imports, that much less module bloat, that many fewer function calls and memory allocations…

(only ranting here because I trust you won't take it harshly and because I trust your opinions!)